src/services/pid_lut/PIDLookupTable_service.h

```diff
@@ -0,0 +1,46 @@
+// Copyright 2024, Nathan Brei
+// Subject to the terms in the LICENSE file found in the top-level directory.
+
+#pragma once
+
+
+#include "PIDLookupTable.h"
+#include <JANA/Services/JServiceLocator.h>
+#include <JANA/JLogger.h>
+#include <mutex>
+#include <filesystem>
+#include <cstdlib>
+
+
+class PIDLookupTable_service : public JService {
+
+    std::mutex m_mutex;
+    std::map<std::string, std::unique_ptr<PIDLookupTable>> m_cache;
+
+public:
+
+    const PIDLookupTable* FetchTable(std::string filename, std::string url) {
+
+        std::lock_guard<std::mutex> lock(m_mutex);
+        auto pair = m_cache.find(filename);
+        if (pair == m_cache.end()) {
+            auto lut = std::make_unique<PIDLookupTable>();
+            LOG << "Loading PID lookup table: " << filename << LOG_END;
+
+            if (!std::filesystem::exists(filename)) {
+                LOG << "PID lookup table '" << filename << "' not found. Attempting to download..." << LOG_END;
+                std::ostringstream oss;
+                oss << "curl -k -o " << filename << " " << url;
+                system(oss.str().c_str());
```

Comment:
Algorithms and services should not run curl, but assume the lookup tables are present. We have artifact distribution mechanisms (however imperfect) through the geometry, and these should be reused instead of adding new interfaces. If the filename does not exist locally, the service should fail.

Reply:
If every service or every algorithm is going to implement its own way to get artifacts, it increases the possibility for errors and our support burden. For operational reasons this is also not a workable solution:
- How will you deal with updates to the LUT? This only checks that the file exists, not that it is the latest version. Anyone running this once in their local development directory will forever be stuck on the first downloaded version.
- How do you deal with this when we run 10k jobs in production? All of them are going to be running in a job-specific directory where the LUT file doesn't exist yet. Are they all going to download the file?

These are not new considerations, but we have solved them for files like Acts material maps.

Reply:
Submitted to https://github.com/eic/epic/pull/711

Reply:
Once that's in nightly, you can expect the `calibrations/hpdirc_positive.lut` to just exist when you run your algorithm.